src/vpu/encoder_h264.c

```diff
@@ -191,6 +193,21 @@ gboolean gst_imx_vpu_encoder_h264_set_open_params(GstImxVpuEncoderBase *vpu_enco
 {
 	GstCaps *template_caps, *allowed_caps;
 	GstImxVpuEncoderH264 *vpu_encoder_h264 = GST_IMX_VPU_ENCODER_H264(vpu_encoder_base);
+	GstVideoFormat fmt = GST_VIDEO_INFO_FORMAT(&(input_state->info));
+
+	if (fmt == GST_VIDEO_FORMAT_GRAY8)
+	{
+		open_params->color_format = IMX_VPU_COLOR_FORMAT_YUV400;
+		if (vpu_encoder_h264->cbcr_buffer == 0)
+		{
+#define CBCR_PLANE_SIZE	(1920 * 1080 / 4)
```

Comment:
Why this macro? Calculate the size based on frame_width and frame_height from the open_params.